tests/providers/google/cloud/operators/test_bigquery.py

```diff
@@ -788,3 +789,65 @@ def test_execute(self, mock_hook):
                 project_id=TEST_GCP_PROJECT_ID,
                 table_resource=TEST_TABLE_RESOURCES
             )
+
+
+class TestBigQueryInsertJobOperator:
+    @mock.patch('airflow.providers.google.cloud.operators.bigquery.BigQueryHook')
+    def test_execute(self, mock_hook):
+        job_id = "123456"
+        configuration = {
+            "query": {
+                "query": "SELECT * FROM any",
+                "useLegacySql": False,
+            }
+        }
+        mock_hook.return_value.insert_job.return_value = MagicMock(job_id=job_id)
+
+        op = BigQueryInsertJobOperator(
+            task_id="insert_query_job",
+            configuration=configuration,
+            location=TEST_DATASET_LOCATION,
+            job_id=job_id,
+            project_id=TEST_GCP_PROJECT_ID
+        )
+        result = op.execute({})
+
+        mock_hook.return_value.insert_job.assert_called_once_with(
+            configuration=configuration,
+            location=TEST_DATASET_LOCATION,
+            job_id=job_id,
+            project_id=TEST_GCP_PROJECT_ID,
+        )
+
+        assert result == job_id
+
+    @mock.patch('airflow.providers.google.cloud.operators.bigquery.BigQueryHook')
```

Comment:
Should we also add a test for retry behaviour?

Reply:
@potiuk what exactly you mean? There are two tests: creating a new job and reattaching to existing one.

Reply:
I mean seeing the retry behavior in practice. Reattaching to existing job succeeds immediately but might be worth (maybe) to have an example where you reattach while the job is still running and you retry once or twice after attaching before you succeed.

Reply:
It's not super needed - nice to have - but I thought it might be useful to add.

Reply:
Added